gearmand/datadog_checks/gearmand/gearmand.py

```diff
@@ -118,3 +118,21 @@ def check(self, instance):
         except Exception as e:
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=str(e), tags=tags)
             raise
+
+        self._collect_metadata(client)
+
+    def _collect_metadata(self, client):
+        try:
```

Comment:
put it behind `is_metadata_collection_enabled` from check base

Reply:
That config option prevents the agent from submitting metadata but is good to also make the integration not collect it in case there are issues with the collecting itself

Reply:
Sarah had a similar comment above see thread here: https://github.com/DataDog/integrations-core/pull/5927/files/66f8d35998b60aefdfa02a6f25e779d7a6fc00a4#diff-efbd850bd09b665a48ae35cefc0045ba

Short version is the Agent stub doesn't support checking the config right now, I'm planning to address that in a separate PR.

Reply:
I've added it here: https://github.com/DataDog/integrations-core/pull/5448/files

Reply:
Ok, I'll wait until that PR is merged, then I'll rebase and incorporate the check.